### DISCUSSION TITLE
Control an accessory by means of the code

### DISCUSSION THREAD

**Discussion by edevs-damien** [HomeSpan/HomeSpan#216] [Q&A] [answered]
Hello,
Is there a way to control a HomeSpan accessory from code (example: from the `loop ()` function) while being able to control it via the home application (on ios). (example: turn on the bulb when a button is pressed or when certain conditions are reached)?

Thank you

Damien

**Comment by HomeSpan** [comment, marked as answer]
Yes, there are many ways to do exactly this. See [Example 12](https://github.com/HomeSpan/HomeSpan/blob/master/docs/Tutorials.md#example-12---serviceloops) for a tutorial on Service Loops (where you can enter any code specific to a Service) as well as [Example 15](https://github.com/HomeSpan/HomeSpan/blob/master/docs/Tutorials.md#example-15---realpushbuttons) that details how to use pushbuttons in conjunction with the Home App to operate a light.

Note you can also enter any code you'd like in the main Arduino loop() function.
